week-2/implement/tally.js

```diff
@@ -0,0 +1,33 @@
+function tally(listOfItems) {
+  if (!Array.isArray(listOfItems)) {
+    throw new Error("Invalid input: Expected an array");
+    // this part added after tests because "throw an error" test did not work
+  }
+
+  let count = {}; // we need to create an empty object to store counts
+
+  for (let i = 0; i < listOfItems.length; i++) {
+    let eachItem = listOfItems[i];
```

Comment:
It's absolutely correct and works perfect! But just a question: do you really need a `for` loop here? Do you really need the indexes? Can you achieve the same result with `for ... of` loop?